Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm

```diff
@@ -77,17 +77,26 @@
 #if USE(EXTENSIONKIT)
 static std::pair<ASCIILiteral, RetainPtr<NSString>> serviceNameAndIdentifier(ProcessLauncher::ProcessType processType, ProcessLauncher::Client* client, bool isRetryingLaunch)
 {
+    bool hasExtensionsInAppBundle = ProcessLauncher::hasExtensionsInAppBundle();
     switch (processType) {
     case ProcessLauncher::ProcessType::Web: {
-        if (client && client->shouldEnableLockdownMode())
+        bool useCaptivePortal = client && client->shouldEnableLockdownMode();
+        if (!hasExtensionsInAppBundle) {
+            if (!useCaptivePortal)
+                return { "com.apple.WebKit.WebContent"_s, @"com.apple.WebKit.WebContent" };
             return { "com.apple.WebKit.WebContent"_s, @"com.apple.WebKit.WebContent.CaptivePortal" };
-        return { "com.apple.WebKit.WebContent"_s, @"com.apple.WebKit.WebContent" };
+        }
+        return { "com.apple.WebKit.WebContent"_s, [NSString stringWithFormat:@"%@.WebContent%@Extension", [[NSBundle mainBundle] bundleIdentifier], (useCaptivePortal ? @"CaptivePortal" : @"")] };
```

Comment:
Perhaps use two format strings instead, `@"%@.WebContentExtension` and `@"%@.WebContentCaptivePortalExtension`?

Reply:
Discussed offline, I actioned this feedback by splitting the ternary into the preceding line and making it more readable.